src/main/src/test/java/org/geoserver/test/GeoServerTestApplicationContext.java

```diff
@@ -96,4 +105,25 @@ protected void initPropertySources() {
         WebApplicationContextUtils.initServletPropertySources(
                 this.getEnvironment().getPropertySources(), this.servletContext);
 	}
+
+    @Override
+    protected void initBeanDefinitionReader(XmlBeanDefinitionReader reader) {
+        super.initBeanDefinitionReader(reader);
+        if (!fullInitRequired) {
```

Comment:
If we can, load the beans lazily instead of fully on startup. This gives a significant benefit during the build, and a very significant one when running the tests on the IDE. 
Reason: each test is only depending on a subset of all the beans, we do not need to initialize them all.